### PR TITLE
Modify all projects to use new RID definition

### PR DESCRIPTION
**Pull request by jaholme**
## Summary of the pull request
This commit removes win10-xxx style RIDs from all csproj. It also updates all publishing profiles to support the .NET 8 style win-xxx RIDS.

## Validation steps performed
Built x86/x64.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Comment by krschau** [comment]
@EricJohnson327 Can you remind me why we did the workarounds in the first place? Was it to play nice with WinAppSDK at the time?

**Comment by manodasanW** [comment]
> @EricJohnson327 Can you remind me why we did the workarounds in the first place? Was it to play nice with WinAppSDK at the time?

If I remember correctly, there are 2 issues which required the need for the RidGraph workaround:

1) We hadn't updated to a Dev Home SDK package that was using the new runtime identifier format
2) The adaptive cards package was also not using the new format and some binaries weren't getting picked up causing adaptive card issues.

We should confirm whether both issues have been addressed or not.

**Comment by krschau** [comment]
> If I remember correctly, there are 2 issues which required the need for the RidGraph workaround:
> 
> 1. We hadn't updated to a Dev Home SDK package that was using the new runtime identifier format
> 2. The adaptive cards package was also not using the new format and some binaries weren't getting picked up causing adaptive card issues.
> 
> We should confirm whether both issues have been addressed or not.

Shoot, widgets don't render with this change. I've opened https://github.com/microsoft/AdaptiveCards/issues/8868.

File: .github/workflows/DevHome-CI.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
     - name: Compress_DevSetupAgent_x86
       if: ${{ matrix.platform != 'arm64' }}
       shell: pwsh
-      run: Compress-Archive -Force -Path HyperVExtension\src\DevSetupAgent\bin\x86\${{ matrix.configuration }}\net8.0-windows10.0.22621.0\win10-x86\* -DestinationPath "HyperVExtension\src\DevSetupAgent\bin\x86\${{ matrix.configuration }}\DevSetupAgent_x86.zip"
+      run: Compress-Archive -Force -Path HyperVExtension\src\DevSetupAgent\bin\x86\${{ matrix.configuration }}\net8.0-windows10.0.22000.0\win-x86\* -DestinationPath "HyperVExtension\src\DevSetupAgent\bin\x86\${{ matrix.configuration }}\DevSetupAgent_x86.zip"
```
Comment (jaholme) [inline .github/workflows/DevHome-CI.yml:75]: need to update these versions.

**Comment by jaholme** [comment]
> > If I remember correctly, there are 2 issues which required the need for the RidGraph workaround:
> > 
> > 1. We hadn't updated to a Dev Home SDK package that was using the new runtime identifier format
> > 2. The adaptive cards package was also not using the new format and some binaries weren't getting picked up causing adaptive card issues.
> > 
> > We should confirm whether both issues have been addressed or not.
> 
> Shoot, widgets don't render with this change. I've opened [microsoft/AdaptiveCards#8868](https://github.com/microsoft/AdaptiveCards/issues/8868).

This is solved with the latest set of changes. DevHome can manually copy the libraries to the package directory as a temporary workaround.

**Review by EricJohnson327 — APPROVED** [review]

**Review by dabhattimsft — APPROVED** [review]

**Pull request merged**